index.js

```diff
@@ -314,3 +305,30 @@ module.exports = fp(
         fastify: '3.x',
     }
 );
+
+function onRouteBind(metrics) {
+    // map id to route url
+    const idToRoute = new Map();
```

Comment:
I prefer to add a closure instead of decorating Fastify instance. If we can not nest this plugin, this approach makes more sense.

Reply:
One nit: we can save `getRouteId` in a variable and call the function just one time. For the closure: I think the Map should be outside of the closure and shared by all plugin registrations. If we use a closure like here we would miss duplicate ids in other plugin instances. wdyt?

Reply:
One other thing,`metrics` here isn't really used, did you plan to use it?

Reply:
Ok :+1:

Reply:
But we can only decorate the fastify instance because if we maintain the Map global we lose the possibility to spawn different instances of fastify in the same code